src/Interpreters/QueryLog.cpp

```diff
@@ -64,16 +64,15 @@ NamesAndTypesList QueryLogElement::getNamesAndTypes()
         {"formatted_query", std::make_shared<DataTypeString>()},
         {"normalized_query_hash", std::make_shared<DataTypeUInt64>()},
         {"query_kind", std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>())},
-        {"databases", std::make_shared<DataTypeArray>(
-            std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
-        {"tables", std::make_shared<DataTypeArray>(
-            std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
-        {"columns", std::make_shared<DataTypeArray>(
-            std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
-        {"projections", std::make_shared<DataTypeArray>(
-            std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
-        {"views", std::make_shared<DataTypeArray>(
-            std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"databases", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"tables", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"columns", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"projections", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"views", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"partitions",
+         std::make_shared<DataTypeMap>(
+             std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()),
```

Comment:
2 open questions (from complete ignorance):
* is `Map(LowCardinality(String), Array(LowCardinality(String))` actually better than not using LC at all? I see that it was used for other types too, but it's unclear to me what  the benefit of using LC inside nested types is.
* Is there a way (maybe impossible, maybe too expensive to be hidden behind an option) to get parts too?

Reply:
I will add a perf test about the first question.

Recording parts is too heavy for system.query_log. In our production environment, a single query may cover above 10k parts which will slow down query execution.